metals/src/main/scala/scala/meta/internal/metals/FileChanges.scala

```diff
@@ -0,0 +1,136 @@
+package scala.meta.internal.metals
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+
+class FileChanges(buildTargets: BuildTargets, workspace: () => AbsolutePath)(
+    implicit ec: ExecutionContext
+) {
+  private val previousSignatures = TrieMap[AbsolutePath, String]()
+  private val dirtyBuildTargets = mutable.Set[BuildTargetIdentifier]()
```

Comment:
```suggestion
  private val invalidatedBuildTargets = mutable.Set[BuildTargetIdentifier]()
```
maybe is a better word?

Reply:
What about this change?